packages/react-components/react-table/src/hooks/useColumnSizing.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import {
+  TableColumnId,
+  ColumnWidthState,
+  TableColumnSizingState,
+  TableFeaturesState,
+  UseColumnSizingParams,
+} from './types';
+import { useColumnResizeState } from './useColumnResizeState';
+import useColumnResizeMouseHandler from './useColumnResizeMouseHandler';
+import { useMeasureElement } from './useMeasureElement';
+import { TableResizeHandle } from '../TableResizeHandle';
+
+export const defaultColumnSizingState: TableColumnSizingState = {
+  getColumnWidth: () => 0,
+  getColumnWidths: () => [],
+  getOnMouseDown: () => () => null,
+  getTotalWidth: () => 0,
+  setColumnWidth: () => null,
+  getColumnProps: () => ({ style: {}, columnId: '' }),
+};
+
+export function useColumnSizing_unstable<TItem>(params?: UseColumnSizingParams) {
+  // False positive, these plugin hooks are intended to be run on every render
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  return (tableState: TableFeaturesState<TItem>) => useColumnSizingState(tableState, params);
+}
+
+function getColumnStyles(column: ColumnWidthState): React.CSSProperties {
+  const width = column.width;
+
+  return {
+    // native styles
+    width,
+    // non-native element styles (flex layout)
+    minWidth: width,
+    maxWidth: width,
+  };
+}
+
+function useColumnSizingState<TItem>(
+  tableState: TableFeaturesState<TItem>,
+  params?: UseColumnSizingParams,
+): TableFeaturesState<TItem> {
+  const { columns } = tableState;
+  const [tableEl, setTableEl] = React.useState<HTMLElement | null>(null);
```

Comment:
Avoid `useState` to set refs to HTML elements since it will force a re-render of the entire component. In this case `useMeasureElement` should actually return a callback ref 

```ts
const { width, measureElementRef } = useMeasureElement();
```

Reply:
I tried to refactor the hook, it seems to work, but you might know more about details I've overlooked

```ts
export function useMeasureElement2<TElement extends HTMLElement = HTMLElement>() {
  const [width, setWidth] = React.useState(0);
  const container = React.useRef<HTMLElement | undefined>(undefined);

  const { targetDocument } = useFluent();

  // the handler for resize observer
  const handleResize = React.useCallback(() => {
    const containerWidth = container.current?.getBoundingClientRect().width;
    setWidth(containerWidth || 0);
  }, []);

  // Keep the reference of ResizeObserver in the state, as it should live through renders
  const [resizeObserver] = React.useState(canUseDOM() ? new ResizeObserver(handleResize) : undefined);
  const measureElementRef = React.useCallback(
    (el: TElement | null) => {
      if (!targetDocument || !resizeObserver) {
        return;
      }

      // cleanup previous container
      if (container.current) {
        resizeObserver.unobserve(container.current);
        container.current.remove();
      }

      if (el) {
        container.current = targetDocument.createElement('div');
        el.insertAdjacentElement('beforebegin', container.current);
        resizeObserver.observe(container.current);
        handleResize();
      }
    },
    [targetDocument, resizeObserver, handleResize],
  );

  React.useEffect(() => {
    return () => resizeObserver?.disconnect();
  }, [resizeObserver]);

  return { width, measureElementRef };
}

```